src/main.rs

```diff
@@ -229,16 +231,33 @@ fn start_program(core: &mut Core, elf: &Elf) -> anyhow::Result<()> {
     Ok(())
 }
 
+fn detach_from_program(core: &mut Core, elf: &Elf) -> anyhow::Result<()> {
+    log::debug!("detaching from program");
+
+    if let Some(rtt_buffer_address) = elf.rtt_buffer_address() {
+        set_rtt_blocking_mode(core, rtt_buffer_address, false)?;
```

Comment:
What if rtt was in non-blocking mode before we attached? Then we probably don't want to set it to blocking afterwards., right?